tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,379 @@
+    }
+
+    @Test(description = "Test running init on an already existing project and create a new package",
+            dependsOnMethods = "testInitWithMainInPackage")
+    public void testInitOnExistingProjectWithNewPackage() throws Exception {
+        // Test ballerina init
+        getNewInstanceOfBallerinaServer();
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesMain");
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "m\n", "newpkg\n", "f\n"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve("newpkg").resolve("main.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("newpkg").resolve("tests").resolve("main_test.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("main.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("tests").resolve("main_test.bal")));
+
+        // Test ballerina build
+        getNewInstanceOfBallerinaServer();
+        ballerinaServer.runMain(new String[0], getEnvVariables(), "build", projectPath.toString());
+        Assert.assertTrue(Files.exists(projectPath.resolve("target").resolve("newpkg.balx")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("target").resolve("foo.balx")));
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina").resolve("repo").resolve(getOrgName())
+                                                  .resolve("foo").resolve("0.0.1").resolve("foo.zip")));
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina").resolve("repo").resolve(getOrgName())
+                                                  .resolve("newpkg").resolve("0.0.1").resolve("newpkg.zip")));
+
+
+        // Test ballerina run on the new package
+        runMainFunction(projectPath, "newpkg");
+
+        // Test ballerina run with balx
+        runMainFunction(projectPath, projectPath.resolve("target").resolve("newpkg.balx").toString());
+    }
+
+    @Test(description = "Test creating a project with invalid options")
+    public void testInitWithInvalidOptions() throws Exception {
+        // Test ballerina init
+        getNewInstanceOfBallerinaServer();
+        Path projectPath = tempProjectDirectory.resolve("testsWithoutPackage");
+        Files.createDirectories(projectPath);
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "123\n", "jkl\n", "f\n"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
```

Comment:
Shouldnt this fail if we are using invalid options ?

Also we need a test case where we try to create a package name with a hyphen. Which is not allowed.

Reply:
If we use invalid options, it doesn't fail. It will prompt that the input provided is invalid and afterwards this will be prompted to the user again "Ballerina source [service/s, main/m, finish/f]: (f) ". Only the .ballerina and Ballerina.toml will be created in such a case